packages/block-editor/src/hooks/utils.js

```diff
@@ -33,7 +33,7 @@ export const cleanEmptyObject = ( object ) => {
 	const cleanedNestedObjects = Object.fromEntries(
 		Object.entries( object )
 			.map( ( [ key, value ] ) => [ key, cleanEmptyObject( value ) ] )
-			.filter( ( [ , value ] ) => Boolean( value ) )
+			.filter( ( [ , value ] ) => value !== undefined )
```

Comment:
Noting that the comment above still says "falsy". We'll need to change that to `undefined` if we keep this behavior.